packages/dev/core/src/FlowGraph/Blocks/Execution/flowGraphSetPropertyBlock.ts

```diff
@@ -36,39 +28,31 @@ export class FlowGraphSetPropertyBlock<ValueT> extends FlowGraphWithOnDoneExecut
      * Input connection: The value to set on the property.
      */
     public readonly value: FlowGraphDataConnection<ValueT>;
+    /**
+     * The component with the templated inputs for the provided path.
+     */
+    public readonly templateComponent: FlowGraphPathComponent;
 
     public constructor(public config: IFlowGraphSetPropertyBlockConfiguration) {
         super(config);
 
-        this.value = this._registerDataInput("value", RichTypeAny);
-        this._registerDataInput(config.subString, RichTypeNumber);
-    }
-
-    private _setProperty(target: any, property: string, value: any): void {
-        const splitProp = property.split(".");
-
-        let currentTarget = target;
-        for (let i = 0; i < splitProp.length - 1; i++) {
-            currentTarget = currentTarget[splitProp[i]];
-        }
-
-        currentTarget[splitProp[splitProp.length - 1]] = value;
+        this.value = this.registerDataInput("value", RichTypeAny);
+        this.templateComponent = new FlowGraphPathComponent(config.path, this);
     }
 
     public _execute(context: FlowGraphContext): void {
-        const target = context._getTargetFromPath(this.config.path, this.config.subString, this);
-        const property = this.config.property;
         const value = this.value.getValue(context);
-
-        if (target && property) {
-            this._setProperty(target, property, value);
-        } else {
-            throw new Error("Invalid target or property");
-        }
+        const path = this.templateComponent.substitutePath(context);
+        path.setProperty(context, value);

```

Comment:
Same as above with `getProperty`. I would add a method to `FlowGraphPathComponent` so that the user only has to make one call to set the property, thus avoiding missing the call to `substitutePath`.